### PR TITLE
Marko support (beta)

### PR DESCRIPTION
**Pull request by steve8708**

### CODE REVIEW

File: packages/core/src/generators/marko/generate.ts [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    str += ` ${key}="${value}" `;
+  }
+  for (const key in json.bindings) {
+    const { code, arguments: cusArgs = ['event'] } = json.bindings[key]!;
+    if (key === '_spread' || key === '_forName') {
+      continue;
+    }
+
+    if (key === 'ref') {
+      str += ` ref=((el) => this.${code} = el) `;
```
Comment (DylanPiercey) [inline packages/core/src/generators/marko/generate.ts:84]: I think this part needs to be tweaked to use the [key](https://markojs.com/docs/class-components/#key) attribute which is the way to gather references in Marko.

It's string based keys (in Marko 5) so you have something like:

```marko
class {
  onMount() {
    console.log("got el", this.getEl("my-key");
  }
}

<div key="my-key"/>
```

File: packages/core/src/generators/marko/generate.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    let jsString = dedent`
+    ${renderPreComponent({ component: json, target: 'marko' })}
+
+    class {
+        ${methodsString}
+
+        ${
+          !hasState
+            ? ''
+            : `onCreate() {
```
Comment (DylanPiercey) [inline packages/core/src/generators/marko/generate.ts:170]: The `onCreate` method receives `input` into it, so you could change this to:

```
`onCreate(${thisHasProps ? 'input' : ''}) {`
```

File: packages/cli/src/build/build.ts [hunk 3]
```diff
@@ -185,6 +186,8 @@ const getGeneratorForTarget = ({
       return componentToSvelte(options.options.svelte);
     case 'qwik':
       return componentToQwik(options.options.qwik);
+    case 'marko':
+      return componentToMarko(options.options.qwik);
```
Comment (samijaber) [inline packages/cli/src/build/build.ts:190]:
```suggestion
      return componentToMarko(options.options.marko);
```

**Review by samijaber — APPROVED** [review]
👍🏽 

just the options error I commented on

File: packages/core/src/generators/marko/generate.ts (hunk 2 above)
Comment (steve8708) [reply packages/core/src/generators/marko/generate.ts:170]: oh perfect, will do that, thanks @DylanPiercey!

File: packages/core/src/generators/marko/generate.ts (hunk 1 above)
Comment (steve8708) [reply packages/core/src/generators/marko/generate.ts:84]: ah! perfect, thanks @DylanPiercey 

I'll remove this old ref logic for now and circle back on a full refs implementation based on this

Comment (steve8708) [reply packages/core/src/generators/marko/generate.ts]: nvm went ahead and implemented, huge thanks for the tip @DylanPiercey this made it very easy to implement

**Pull request merged**